pkg/controller/notification/notification_controller.go

```diff
@@ -166,12 +186,46 @@ func (r *ReconcileNotification) setStatusNotificationDeletionFailed(notification
 		})
 }
 
-func (r *ReconcileNotification) setStatusNotificationSent(notification *toolchainv1alpha1.Notification) error {
+func (r *ReconcileNotification) setStatusNotificationContextError(notification *toolchainv1alpha1.Notification, msg string) error {
 	return r.updateStatusConditions(
 		notification,
 		toolchainv1alpha1.Condition{
-			Type:   toolchainv1alpha1.NotificationSent,
-			Status: corev1.ConditionTrue,
-			Reason: toolchainv1alpha1.NotificationSentReason,
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NotificationContextErrorReason,
+			Message: msg,
 		})
 }
+
+func (r *ReconcileNotification) setStatusNotificationDeliveryError(notification *toolchainv1alpha1.Notification, msg string) error {
+	return r.updateStatusConditions(
+		notification,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NotificationDeliveryErrorReason,
+			Message: msg,
+		})
+}
+
+func (r *ReconcileNotification) setStatusNotificationSent(notification *toolchainv1alpha1.Notification, msg string) error {
+	return r.updateStatusConditions(
+		notification,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionTrue,
+			Reason:  toolchainv1alpha1.NotificationSentReason,
+			Message: msg,
+		})
+}
+
+func (r *ReconcileNotification) updateStatus(logger logr.Logger, notification *toolchainv1alpha1.Notification,
+	statusUpdater func(notification *toolchainv1alpha1.Notification, message string) error) error {
+
+	if err := statusUpdater(notification, ""); err != nil {
```

Comment:
Hmm, good question... this function was copied from the UserSignupController.... @alexeykazakov do you know why we pass empty string for the message?

Reply:
`message` is only used when we catch an error. It's for the error message. And since we handle errors by `wrapErrorWithStatusUpdate()` all other status updates don't need messages. So, I think we can remove `message` from statusUpdater function to avoid confusion.

Reply:
I'm not so sure it's easy to do this.  Each of the status updater functions accept a message parameter and uses it to set the message in the condition, like so:

````
func (r *ReconcileNotification) setStatusNotificationDeliveryError(notification *toolchainv1alpha1.Notification, msg string) error {
	return r.updateStatusConditions(
		notification,
		toolchainv1alpha1.Condition{
			Type:    toolchainv1alpha1.NotificationSent,
			Status:  corev1.ConditionFalse,
			Reason:  toolchainv1alpha1.NotificationDeliveryErrorReason,
			Message: msg,
		})
}
````

If I remove the `message` parameter from the `statusUpdater` function then I'll also need to remove this parameter from all of the status update functions.  Is that the intent here?

Reply:
@sbryzak if `message` is actually used by some function then we have to keep it. If not then we can delete it.